src/utils/regex.cc

```diff
@@ -311,5 +378,43 @@ int Regex::search(const std::string& s) const {
 #endif
 }
 
+unsigned long Regex::get_default_match_limit() const {
+    unsigned long default_match_limit;
+#ifdef WITH_PCRE2
+    int ret = pcre2_config(PCRE2_CONFIG_MATCHLIMIT, &default_match_limit);
+#else
+    int ret = pcre_config(PCRE_CONFIG_MATCH_LIMIT, &default_match_limit);
+#endif
+    if (ret < 0) {
+        default_match_limit = 10000000;
+    }
+    return default_match_limit;
+}
+
+RegexResult Regex::to_regex_result(int pcre_exec_result) const {
```

Comment:
cppcheck may complain about this.
It doesn't do anything with an object's state (e.g. no member variables are accessed), it could be a static function.

Don't worry about this for now.